restore-hooks_product-requirements.md

```diff
@@ -0,0 +1,376 @@
+
+**<span style="text-decoration:underline;">USE CASE 2</span>**
+
+
+    **Title: **Allow restore hook to run on non-kubernetes databases
+
+
+    **Description: **As a user, I would like to run restore hook operations even on databases that are external to kubernetes (such as postgres, elastic, etc…).
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 3</span>**
+
+
+    **Title: **Run restore at pod level.
+
+
+    **Description: **As a user, I would like to make sure that I can run the restore hook at the pod level. And, I would like the option to run this with an annotation flag in line or using an init container.**<span style="text-decoration:underline;"> </span>**The restore pre-hook should allow the user to run the command on the container where the pre-hook should be executed. Similar to the backup hooks, this hook should run to default to fun on the first container in the pod. 
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 4</span>**
+
+
+    **Title:** Specify container in which to run pre-hook
+
+
+    **Description: **As a user, if I do not want to run the pre-hook command on the first container in the pod (default container), I would like the option to annotate the specific container that i would like the hook to run in.
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 5</span>**
+
+
+    **Title: **Check for latest snapshot 
+
+
+    **Description: **As a user, I would like Velero to run a check for the latest snapshot in object storage prior to starting restore operations on a pod. 
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 6</span>**
+
+
+    **Title: **Specify backup snapshot to use for restore
+
+
+    **Description: **As a user, I would like the option to specify a specific backup snapshot for use by Velero during restore create. I would like to do this instead of using the default most recent backup restore snapshot.
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 7</span>**
+
+
+    **Title: **Include or exclude namespaces from restore
+
+
+    **Description: **As a user, I would like to include all resources in namespaces contained in a backup  by default in my restore spec. 
+
+
+    If velero is asked to restore something that already exists in a pod, the restore will not return a success but will still work - Ashish needs to verify.
```

Comment:
Is this meant to extend to the restore hooks themselves? So if the restore hook tries to restore a directory that is already populated within a pod, are we saying that the hook should let Velero know that it happened, but not fail the whole backup?

Reply:
Not sure I understand how this relates to restore hooks. I thought restore hooks lets the user do something after the pod had been restored but not repeat a restore action

Reply:
I am actually going to remove this use case from the PRD. I see in the backup hooks API type documentation, that a user can specify the array of resources to include or exclude from a backup. The intent of this requirement was to address a use case of that nature. 

Because that is already addressed as a capability of the existing Velero backup procedures and is something that would occur prior to restore operations. As such, removing this makes sense and does not confuse the restore hooks actions. Thoughts:  @nrb @yaronp68? (reference:  https://github.com/vmware-tanzu/velero/edit/stephbman-patch-1/restore-hooks_product-requirements.md?pr=%2Fvmware-tanzu%2Fvelero%2Fpull%2F2699).

Thank you for the feedback as well!

Reply:
Yeah, I think removing it makes sense.